scio-core/src/main/scala/com/spotify/scio/coders/JTraversableSerializer.scala

```diff
@@ -77,19 +75,41 @@ private class JIterableWrapperSerializer[T](val bufferSize: Int = 64 * 1024,
     }
   }
 
-  override def read(kser: Kryo, in: Input, cls: Class[Iterable[T]]): Iterable[T] = {
-    val list = Lists.newArrayList[T]
+  override def read(kser: Kryo, in: Input, cls: Class[C]): C = {
+    val b = cbf()
     var count = in.readInt()
     while (count > 0) {
       var i = 0
       while (i < count) {
         val item = kser.readClassAndObject(in).asInstanceOf[T]
-        list.add(item)
+        b += item
         i += 1
       }
       count = in.readInt()
     }
-    list.asInstanceOf[JIterable[T]].asScala
+    b.result()
   }
 
 }
+
+// workaround for Iterable missing proper equality check
+private class JIterableWrapperCBF[T] extends CanBuildFrom[Iterable[T], T, Iterable[T]] {
```

Comment:
Tests missing for `JIterableWrapperCBF`

Reply:
It's actually covered in `KryoAtomicCoderTest` but I'm gonna add a few more for other cases.